app/components/wallet/trezorConnect/SaveDialog.js

```diff
@@ -0,0 +1,189 @@
+    defaultMessage: '!!!Enter wallet name',
+    description: 'Placeholder "Enter wallet name" for the wallet name input on the wallet restore dialog.'
+  },
+  saveWalletNameInputBottomInfo: {
+    id: 'wallet.trezor.dialog.trezor.step.save.walletName.info',
+    defaultMessage: '!!!We have fetched Trezor device’s name for you; you can use as it is or assign a different name.',
+    description: 'Hint for the wallet name input on the wallet restore dialog.'
+  },
+  saveButtonLabel: {
+    id: 'wallet.trezor.dialog.trezor.save.button.label',
+    defaultMessage: '!!!Save',
+    description: 'Label for the "Save" button on the Connect to Trezor Hardware Wallet dialog.'
+  },
+});
+
+messages.fieldIsRequired = globalMessages.fieldIsRequired;
+
+type Props = {
+  progressInfo: ProgressInfo,
+  error: ?LocalizableError,
+  isActionProcessing: boolean,
+  defaultWalletName: string,
+  submit: Function,
+  cancel: Function,
+};
+
+@observer
+export default class SaveDialog extends Component<Props> {
+
+  static contextTypes = {
+    intl: intlShape.isRequired
+  };
+
+  // form for wallet name
+  form: typeof ReactToolboxMobxForm;
+
+  componentWillMount() {
+    const { intl } = this.context;
+
+    this.form = new ReactToolboxMobxForm({
+      fields: {
+        walletName: {
+          label: intl.formatMessage(messages.saveWalletNameInputLabel),
+          placeholder: intl.formatMessage(messages.saveWalletNameInputPlaceholder),
+          value: this.props.defaultWalletName,
+          validators: [({ field }) => (
+            [
+              isValidHardwareWalletName(field.value),
+              intl.formatMessage(globalMessages.invalidHardwareWalletName)
+            ]
+          )],
+        },
+      },
+    }, {
+      options: {
+        validateOnChange: true,
+        validationDebounceWait: 250,
+      },
+    });
+  }
+
+  render() {
+    const { intl } = this.context;
+
+    // walletNameBlock
+    const walletNameFieldClasses = classnames([
+      'walletName',
+      styles.walletName,
+    ]);
+    const walletNameField = this.form.$('walletName');
+
+    const walletNameBlock = (
+      <div className={classnames([styles.headerBlock, styles.headerSaveBlock])}>
+        <Input
+          className={walletNameFieldClasses}
+          {...walletNameField.bind()}
+          error={walletNameField.error}
+          skin={<SimpleInputSkin />}
+        />
+        <span>{intl.formatMessage(messages.saveWalletNameInputBottomInfo)}</span>
+      </div>);
+
+    // middleBlock selection depending upon state
+    let middleBlock = null;
+
+    switch (this.props.progressInfo.stepState) {
+      case StepStateOption.LOAD:
+        middleBlock = (
+          <div className={classnames([styles.middleBlock, styles.middleSaveLoadBlock])}>
+            <SvgInline svg={saveLoadGIF} cleanup={['title']} />
+          </div>);
+        break;
+      case StepStateOption.PROCESS:
+        // START
```

Comment:
I feel this comment should be more verbose (if it’s actually something complex to describe in more than one word) or otherwise I would change the name of the enum to make it more obvious

Reply:
before `PROCESS` state was `START`state 
I forget to delete the comment line.

will delete  `// START` because there is nothing complex